packages/dataclass-editor/src/detail/field/ComboboxFieldWithTypeBrowser.tsx

```diff
@@ -0,0 +1,78 @@
+import {
+  BasicCheckbox,
+  BasicField,
+  Button,
+  Combobox,
+  Dialog,
+  DialogContent,
+  DialogTrigger,
+  Flex,
+  IvyIcon,
+  type BrowserNode,
+  type MessageData
+} from '@axonivy/ui-components';
+import { IvyIcons } from '@axonivy/ui-icons';
+import { useEffect, useMemo, useState } from 'react';
+import { Browser } from './browser/Browser';
+import { useAppContext } from '../../context/AppContext';
+import { useMeta } from '../../context/useMeta';
+import { typeData } from '../../data/type-data';
+import './ComboboxFieldWithTypeBrowser.css';
+import type { DataclassType } from '../../protocol/types';
+
+export type InputFieldProps = {
+  value: string;
+  onChange: (value: string) => void;
+  message?: MessageData;
+};
+
+export const ComboboxFieldWithTypeBrowser = ({ value, onChange, message }: InputFieldProps) => {
+  const [open, setOpen] = useState(false);
+  const { context } = useAppContext();
+  const [typeAsList, setTypeAsList] = useState<boolean>(false);
+  const dataClasses = useMeta('meta/scripting/dataClasses', context, []).data;
+  const ivyTypes = useMeta('meta/scripting/ivyTypes', undefined, []).data;
+  const types = useMemo(() => typeData(dataClasses, ivyTypes, [], [], false), [dataClasses, ivyTypes]);
+
+  useEffect(() => {
+    if (typeAsList && !(value.startsWith('List<') && value.endsWith('>'))) {
+      onChange(`List<${value}>`);
+    } else if (!typeAsList && value.startsWith('List<') && value.endsWith('>')) {
+      const removeListFromValue = value.slice(5, -1);
+      onChange(removeListFromValue);
+    }
+  }, [onChange, typeAsList, value]);
```

Comment:
why is a useEffect needed here?
Why not create a function which is triggered when the checkbox changes?

```tsx
typeAsListChange = () => {
  setTypeASList(!typeAsList);
  if (typeAsList && !(value.startsWith('List<') && value.endsWith('>'))) {
    onChange(`List<${value}>`);
  } else if (!typeAsList && value.startsWith('List<') && value.endsWith('>')) {
    const removeListFromValue = value.slice(5, -1);
    onChange(removeListFromValue);
  }
};
  
<BasicCheckbox label='Type as List' checked={typeAsList} onCheckedChange={typeAsListChange} />
```

Reply:
maybe it is even not needed to hold the typeAsList in a state? just only call this function if it changes?

```tsx
<BasicCheckbox label='Type as List' onCheckedChange={change => typeAsListChange(change)} />
```

Reply:
I believe I need to use state, because I automatically set the Checkbox when a List-Type is entered via the browser. I think this can only be achieved using useState, correct?

Reply:
ah didn't saw that. yes then I think too the state is needed.